pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -1016,27 +1016,27 @@ public CompletableFuture<Optional<Topic>> getTopic(final TopicName topicName, bo
                 }
             }
             final boolean isPersistentTopic = topicName.getDomain().equals(TopicDomain.persistent);
-            if (isPersistentTopic) {
-                return topics.computeIfAbsent(topicName.toString(), (k) -> {
-                    return this.loadOrCreatePersistentTopic(k, createIfMissing, properties);
-                });
-            } else {
-                return topics.computeIfAbsent(topicName.toString(), (name) -> {
+            return topics.computeIfAbsent(topicName.toString(), (name) -> {
+                    // partitioned topic
                     if (topicName.isPartitioned()) {
                         final TopicName partitionedTopicName = TopicName.get(topicName.getPartitionedTopicName());
-                        return this.fetchPartitionedTopicMetadataAsync(partitionedTopicName).thenCompose((metadata) -> {
+                        return fetchPartitionedTopicMetadataAsync(partitionedTopicName).thenCompose((metadata) -> {
                             if (topicName.getPartitionIndex() < metadata.partitions) {
-                                return createNonPersistentTopic(name);
+                                return isPersistentTopic ?
+                                        loadOrCreatePersistentTopic(name, createIfMissing, properties) :
```

Comment:
Good work! I just have a little question.
If the `partitionIndex` is 0 and the metadata can not be found, what will happen here?

Reply:
Seem will throw TopicNotFound

Reply:
>If the partitionIndex is 0 and the metadata can not be found, what will happen here?

I think we should reject the topic creation if it includes `partition` keywords but it didn't have partitioned metadata.

Reply:
@liangyepianzhou  

Currently, pulsar already supports create non-partitioned persistent topic that name with `-partition-`, So, I will keep the logic in this PR, and start discussion in the future.